xbmc/utils/HevcSei.cpp

```diff
@@ -0,0 +1,187 @@
+  {
+    last_payload_size_byte = br.ReadBits(8);
+    sei.payload_size += 255;
+  }
+
+  sei.payload_size += last_payload_size_byte;
+  sei.payload_offset = br.Position() / 8;
+
+  // Invalid size
+  if (sei.payload_size > br.AvailableBits())
+    return 1;
+
+  br.SkipBits(sei.payload_size * 8);
+  messages.push_back(sei);
+
+  return 0;
+}
+
+std::vector<CHevcSei> CHevcSei::ParseSeiRbspInternal(const uint8_t* buf, const size_t len)
+{
+  std::vector<CHevcSei> messages;
+
+  if (len > 4)
+  {
+    CBitstreamReader br(buf, len);
+
+    // forbidden_zero_bit, nal_type, nuh_layer_id, temporal_id
+    // nal_type == SEI_PREFIX should already be verified by caller
+    br.SkipBits(16);
+
+    while (true)
+    {
+      if (ParseSeiMessage(br, messages))
+        break;
+
+      if (br.AvailableBits() <= 8)
+        break;
+    }
+  }
+
+  return messages;
+}
+
+std::vector<CHevcSei> CHevcSei::ParseSeiRbsp(const uint8_t* buf, const size_t len)
+{
+  return ParseSeiRbspInternal(buf, len);
+}
+
+std::vector<CHevcSei> CHevcSei::ParseSeiRbspUnclearedEmulation(const uint8_t* in_data,
+                                                                     const size_t in_data_len,
+                                                                     std::vector<uint8_t>& buf)
+{
+  hevc_clear_start_code_emulation_prevention_3_byte(in_data, in_data_len, buf);
+  return ParseSeiRbsp(buf.data(), buf.size());
+}
+
+std::optional<const CHevcSei*> CHevcSei::FindHdr10PlusSeiMessage(
+    const std::vector<uint8_t>& buf, const std::vector<CHevcSei>& messages)
+{
+  for (const CHevcSei& sei : messages)
+  {
+    // User Data Registered ITU-T T.35
+    if (sei.payload_type == 4 && sei.payload_size >= 7)
+    {
+      CBitstreamReader br(buf.data() + sei.payload_offset, sei.payload_size);
+      const auto itu_t_t35_country_code = br.ReadBits(8);
+      const auto itu_t_t35_terminal_provider_code = br.ReadBits(16);
+      const auto itu_t_t35_terminal_provider_oriented_code = br.ReadBits(16);
+
+      // United States, Samsung Electronics America, ST 2094-40
+      if (itu_t_t35_country_code == 0xB5 && itu_t_t35_terminal_provider_code == 0x003C &&
+          itu_t_t35_terminal_provider_oriented_code == 0x0001)
+      {
+        const auto application_identifier = br.ReadBits(8);
+        const auto application_version = br.ReadBits(8);
+
+        if (application_identifier == 4 && application_version <= 1)
+          return &sei;
+      }
+    }
+  }
+
+  return {};
+}
+
+std::pair<bool, const std::vector<uint8_t>> CHevcSei::RemoveHdr10PlusFromSeiNalu(
+    const uint8_t* in_data, const size_t in_data_len)
+{
+  bool containsHdr10Plus = false;
+
+  std::vector<uint8_t> buf;
+  std::vector<CHevcSei> messages =
+      CHevcSei::ParseSeiRbspUnclearedEmulation(in_data, in_data_len, buf);
+
+  if (auto res = CHevcSei::FindHdr10PlusSeiMessage(buf, messages))
+  {
+    auto msg = *res;
+
+    containsHdr10Plus = true;
+    if (messages.size() > 1)
```

Comment:
```suggestion
    if (!messages.isEmpty())
```

Reply:
Not the same, it looks for > 1.

Reply:
Alright!

Reply:
Do you mean `messages.empty()` ?

Anyway is not equivalent (`!empty()` is same as `> 0` but code condition is > 1)